### ISSUE TITLE
msgflo-foreign-participant: Uses a lot of RAM

### ISSUE THREAD

**Issue by jonnor** [msgflo/msgflo#74]
Grows to around 100 MB before settling down to 50MB. This means that even small systems with say 10 foreign participants can't really be run on systems with limited amounts of memory (like 512MB Rapsberry PI 1).

< 10 MB would be acceptable.

![msgflo-foreigh-ramuse](https://cloud.githubusercontent.com/assets/45185/22525728/c33a0306-e8c8-11e6-896a-bbba91821a03.png)

**Comment by bergie** [comment]
For memory-constrained devices it would also be good to allow registering multiple foreign participants with a single register process

**Comment by jonnor** [comment]
Yes, can be done as special-case optimization. But not easy to do with component/handler model, as it expects to run a single command for each role to instantiate.

**Comment by jonnor** [comment]
On my computer just opening a `node` REPL takes 25 MB of ram (resident). Even tried pressuring it with `--max-old-size`, did not go lower. So too get this properly down would need to implement it in another language.

For ref, a `python2` REPL was 7 MB. A static binary made with Go or Rust would hopefully be even lower.

**Comment by bergie** [comment]
Proposed solution is to implement msgflo-register-foreign compatible Rust tool: msgflo/msgflo-rust#7
